### ISSUE TITLE
Styling false negative and false positives around "private" and "protected" class members

### ISSUE THREAD

**Issue by maciejmatuszak** [PyCQA/pylint#6568]
### Bug description

I am trying to use `<>-naming-style=<>` syntax in pylintrc file, however there seems to be couple problems around "private" and "protected" class/instance members. 
According to [PEP8](https://peps.python.org/pep-0008/#descriptive-naming-styles)  leading underscore(s) are special case most commonly described as protected and private members: 

> In addition, the following special forms using leading or trailing underscores are recognized (these can generally be combined with any case convention):
>_single_leading_underscore: weak “internal use” indicator. E.g. from M import * does not import objects whose names start with an underscore.
>single_trailing_underscore_: used by convention to avoid conflicts with Python keyword, e.g.
> tkinter.Toplevel(master, class_='ClassName')
> __double_leading_underscore: when naming a class attribute, invokes name mangling (inside class FooBar, __boo becomes _FooBar__boo; see below).
> __double_leading_and_trailing_underscore__: “magic” objects or attributes that live in user-controlled namespaces. E.g. __init__, __import__ or __file__. Never invent such names; only use them as documented.

From the behaviour exhibited by pylint it looks it threat the leading underscore as lower case letter instead of "access modifier" as intended by PEP8. 
The config supplied is default generated by pylint with some `<>-naming-styles` styles modified and disabled irrelevant warnings. 
See the lines I marked as `ERROR:` below

```python
"""
Module doc string
"""

from enum import Enum


def module_function():  # should reject
    pass


def moduleFunction():  # should be OK
    pass


def ModuleFunction():  # should reject
    pass


class SampleEnum(Enum):
    """
    Sample Enum
    """
    CAPS_CASE = 0  # should be OK

    snake_case = 1  # should reject
    _snake_case = 2  # should reject
    __snake_case = 3  # should reject

    camelCase = 4  # should reject
    _camelCase = 5  # should reject
    __camelCase = 6  # should reject

    PascalCase = 7  # should reject
    _PascalCase = 8  # should reject
    __PascalCase = 9  # should reject


class CaseTestClass:
    """
    case testing
    """
    # class-attribute-naming-style=PascalCase all others should be rejected
    class_attr_snake_case = 0  # should reject
    _class_attr_snake_case = 0  # should reject
    __class_attr_snake_case = 0  # should reject

    classAttrCamelCase = 0  # should reject
    _classAttrCamelCase = 0  # should reject - ERROR: this is "protected" class attribute in camelCase instead of PascalCase
    __classAttrCamelCase = 0  # should reject

    ClassAttrPascalCase = 0  # should be OK
    _ClassAttrPascalCase = 0  # should be OK
    __ClassAttrPascalCase = 0  # should be OK - ERROR: this is private class attribute in PascalCase it should be ok but generates invalid-name

    # there is no specific styling for "class methods" so assume "method" styling applies
    # method-naming-style=camelCase
    @classmethod
    def class_method_snake_case(cls):  # should reject
        pass

    @classmethod
    def _class_method_snake_case(cls):  # should reject
        pass

    @classmethod
    def __class_method_snake_case(cls):  # should reject
        pass

    @classmethod
    def classMethodCamelCase(cls):  # should be OK
        pass

    @classmethod
    def _classMethodCamelCase(cls):  # should be OK
        pass

    @classmethod
    def __classMethodCamelCase(cls):  # should be OK - ERROR: this is private class method in camelCase - should be OK but generates invalid-name
        pass

    @classmethod
    def ClassMethodPascalCase(cls):  # should reject
        pass

    @classmethod
    def _ClassMethodPascalCase(cls):  # should reject - ERROR: this is "protected" method in PascalCase - it should be rejected
        pass

    @classmethod
    def __ClassMethodPascalCase(cls):  # should reject
        pass

    def __init__(self) -> None:
        super().__init__()
        self.snake_case_attr = 0  # should be OK
        self._snake_case_attr = 0  # should be OK
        self.__snake_case_attr = 0  # should be OK

        self.camelCaseAttr = 0  # should reject
        self._camelCaseAttr = 0  # should reject
        self.__camelCaseAttr = 0  # should reject

        self.PascalCaseAttr = 0  # should reject
        self._PascalCaseAttr = 0  # should reject
        self.__PascalCaseAttr = 0  # should reject

    def snake_case_method(self):  # should reject

        # use attributes to avoid "unused <>" warnings
        print(self.__snake_case_attr)
        print(self.__camelCaseAttr)
        print(self.__PascalCaseAttr)

        CaseTestClass.__class_method_snake_case()
        CaseTestClass.__classMethodCamelCase()
        CaseTestClass.__ClassMethodPascalCase()

        self.__snake_case_method()
        self.__camelCaseMethod()
        self.__PascalCaseMethod()

    def _snake_case_method(self):  # should reject
        pass

    def __snake_case_method(self):  # should reject
        pass

    def camelCaseMethod(self):  # should be OK
        pass

    def _camelCaseMethod(self):  # should be OK
        pass

    def __camelCaseMethod(self):  # should be OK - ERROR: this is private method in camelCase should be OK but generates invalid-name
        pass

    def PascalCaseMethod(self):  # should reject
        pass

    def _PascalCaseMethod(self):  # should reject - ERROR: this is "protected" method in PascalCase should be rejected but it is not
        pass

    def __PascalCaseMethod(self):  # should reject
        pass

```

### Configuration

```ini
[MASTER]

# A comma-separated list of package or module names from where C extensions may
# be loaded. Extensions are loading into the active Python interpreter and may
# run arbitrary code.
extension-pkg-allow-list=

# A comma-separated list of package or module names from where C extensions may
# be loaded. Extensions are loading into the active Python interpreter and may
# run arbitrary code. (This is an alternative name to extension-pkg-allow-list
# for backward compatibility.)
extension-pkg-whitelist=

# Return non-zero exit code if any of these messages/categories are detected,
# even if score is above --fail-under value. Syntax same as enable. Messages
# specified are enabled, while categories only check already-enabled messages.
fail-on=

# Specify a score threshold to be exceeded before program exits with error.
fail-under=10.0

# Files or directories to be skipped. They should be base names, not paths.
ignore=CVS

# Add files or directories matching the regex patterns to the ignore-list. The
# regex matches against paths and can be in Posix or Windows format.
ignore-paths=

# Files or directories matching the regex patterns are skipped. The regex
# matches against base names, not paths.
ignore-patterns=

# Python code to execute, usually for sys.path manipulation such as
# pygtk.require().
#init-hook=

# Use multiple processes to speed up Pylint. Specifying 0 will auto-detect the
# number of processors available to use.
jobs=1

# Control the amount of potential inferred values when inferring a single
# object. This can help the performance when dealing with large functions or
# complex, nested conditions.
limit-inference-results=100

# List of plugins (as comma separated values of python module names) to load,
# usually to register additional checkers.
load-plugins=

# Pickle collected data for later comparisons.
persistent=yes

# Minimum Python version to use for version dependent checks. Will default to
# the version used to run pylint.
py-version=3.8

# When enabled, pylint would attempt to guess common misconfiguration and emit
# user-friendly hints instead of false-positive error messages.
suggestion-mode=yes

# Allow loading of arbitrary C extensions. Extensions are imported into the
# active Python interpreter and may run arbitrary code.
unsafe-load-any-extension=no


[MESSAGES CONTROL]

# Only show warnings with the listed confidence levels. Leave empty to show
# all. Valid levels: HIGH, INFERENCE, INFERENCE_FAILURE, UNDEFINED.
confidence=

# Disable the message, report, category or checker with the given id(s). You
# can either give multiple identifiers separated by comma (,) or put this
# option multiple times (only on the command line, not in the configuration
# file where it should appear only once). You can also use "--disable=all" to
# disable everything first and then reenable specific checks. For example, if
# you want to run only the similarities checker, you can use "--disable=all
# --enable=similarities". If you want to run only the classes checker, but have
# no Warning level messages displayed, use "--disable=all --enable=classes
# --disable=W".
disable=raw-checker-failed,
        bad-inline-option,
        locally-disabled,
        file-ignored,
        suppressed-message,
        useless-suppression,
        deprecated-pragma,
        use-symbolic-message-instead,
        missing-function-docstring,

# Enable the message, report, category or checker with the given id(s). You can
# either give multiple identifier separated by comma (,) or put this option
# multiple time (only on the command line, not in the configuration file where
# it should appear only once). See also the "--disable" option for examples.
enable=c-extension-no-member


[REPORTS]

# Python expression which should return a score less than or equal to 10. You
# have access to the variables 'error', 'warning', 'refactor', and 'convention'
# which contain the number of messages in each category, as well as 'statement'
# which is the total number of statements analyzed. This score is used by the
# global evaluation report (RP0004).
evaluation=10.0 - ((float(5 * error + warning + refactor + convention) / statement) * 10)

# Template used to display messages. This is a python new-style format string
# used to format the message information. See doc for all details.
#msg-template=

# Set the output format. Available formats are text, parseable, colorized, json
# and msvs (visual studio). You can also give a reporter class, e.g.
# mypackage.mymodule.MyReporterClass.
output-format=text

# Tells whether to display a full report or only the messages.
reports=no

# Activate the evaluation score.
score=yes


[REFACTORING]

# Maximum number of nested blocks for function / method body
max-nested-blocks=5

# Complete name of functions that never returns. When checking for
# inconsistent-return-statements if a never returning function is called then
# it will be considered as an explicit return statement and no message will be
# printed.
never-returning-functions=sys.exit,argparse.parse_error


[FORMAT]

# Expected format of line ending, e.g. empty (any line ending), LF or CRLF.
expected-line-ending-format=

# Regexp for a line that is allowed to be longer than the limit.
ignore-long-lines=^\s*(# )?<?https?://\S+>?$

# Number of spaces of indent required inside a hanging or continued line.
indent-after-paren=4

# String used as indentation unit. This is usually "    " (4 spaces) or "\t" (1
# tab).
indent-string='    '

# Maximum number of characters on a single line.
max-line-length=100

# Maximum number of lines in a module.
max-module-lines=1000

# Allow the body of a class to be on the same line as the declaration if body
# contains single statement.
single-line-class-stmt=no

# Allow the body of an if to be on the same line as the test if there is no
# else.
single-line-if-stmt=no


[SPELLING]

# Limits count of emitted suggestions for spelling mistakes.
max-spelling-suggestions=4

# Spelling dictionary name. Available dictionaries: none. To make it work,
# install the 'python-enchant' package.
spelling-dict=

# List of comma separated words that should be considered directives if they
# appear and the beginning of a comment and should not be checked.
spelling-ignore-comment-directives=fmt: on,fmt: off,noqa:,noqa,nosec,isort:skip,mypy:

# List of comma separated words that should not be checked.
spelling-ignore-words=

# A path to a file that contains the private dictionary; one word per line.
spelling-private-dict-file=

# Tells whether to store unknown words to the private dictionary (see the
# --spelling-private-dict-file option) instead of raising a message.
spelling-store-unknown-words=no


[TYPECHECK]

# List of decorators that produce context managers, such as
# contextlib.contextmanager. Add to this list to register other decorators that
# produce valid context managers.
contextmanager-decorators=contextlib.contextmanager

# List of members which are set dynamically and missed by pylint inference
# system, and so shouldn't trigger E1101 when accessed. Python regular
# expressions are accepted.
generated-members=

# Tells whether missing members accessed in mixin class should be ignored. A
# class is considered mixin if its name matches the mixin-class-rgx option.
ignore-mixin-members=yes

# Tells whether to warn about missing members when the owner of the attribute
# is inferred to be None.
ignore-none=yes

# This flag controls whether pylint should warn about no-member and similar
# checks whenever an opaque object is returned when inferring. The inference
# can return multiple potential results while evaluating a Python object, but
# some branches might not be evaluated, which results in partial inference. In
# that case, it might be useful to still emit no-member and other checks for
# the rest of the inferred objects.
ignore-on-opaque-inference=yes

# List of class names for which member attributes should not be checked (useful
# for classes with dynamically set attributes). This supports the use of
# qualified names.
ignored-classes=optparse.Values,thread._local,_thread._local

# List of module names for which member attributes should not be checked
# (useful for modules/projects where namespaces are manipulated during runtime
# and thus existing member attributes cannot be deduced by static analysis). It
# supports qualified module names, as well as Unix pattern matching.
ignored-modules=

# Show a hint with possible names when a member name was not found. The aspect
# of finding the hint is based on edit distance.
missing-member-hint=yes

# The minimum edit distance a name should have in order to be considered a
# similar match for a missing member name.
missing-member-hint-distance=1

# The total number of similar names that should be taken in consideration when
# showing a hint for a missing member.
missing-member-max-choices=1

# Regex pattern to define which classes are considered mixins ignore-mixin-
# members is set to 'yes'
mixin-class-rgx=.*[Mm]ixin

# List of decorators that change the signature of a decorated function.
signature-mutators=


[BASIC]

# Naming style matching correct argument names.
argument-naming-style=snake_case

# Regular expression matching correct argument names. Overrides argument-
# naming-style.
#argument-rgx=

# Naming style matching correct attribute names.
attr-naming-style=snake_case

# Regular expression matching correct attribute names. Overrides attr-naming-
# style.
#attr-rgx=

# Bad variable names which should always be refused, separated by a comma.
bad-names=foo,
          bar,
          baz,
          toto,
          tutu,
          tata

# Bad variable names regexes, separated by a comma. If names match any regex,
# they will always be refused
bad-names-rgxs=

# Naming style matching correct class attribute names.
class-attribute-naming-style=PascalCase

# Regular expression matching correct class attribute names. Overrides class-
# attribute-naming-style.
#class-attribute-rgx=

# Naming style matching correct class constant names.
class-const-naming-style=UPPER_CASE

# Regular expression matching correct class constant names. Overrides class-
# const-naming-style.
#class-const-rgx=

# Naming style matching correct class names.
class-naming-style=PascalCase

# Regular expression matching correct class names. Overrides class-naming-
# style.
#class-rgx=

# Naming style matching correct constant names.
const-naming-style=UPPER_CASE

# Regular expression matching correct constant names. Overrides const-naming-
# style.
#const-rgx=

# Minimum line length for functions/classes that require docstrings, shorter
# ones are exempt.
docstring-min-length=-1

# Naming style matching correct function names.
function-naming-style=camelCase

# Regular expression matching correct function names. Overrides function-
# naming-style.
#function-rgx=

# Good variable names which should always be accepted, separated by a comma.
good-names=i,
           j,
           k,
           ex,
           Run,
           _

# Good variable names regexes, separated by a comma. If names match any regex,
# they will always be accepted
good-names-rgxs=

# Include a hint for the correct naming format with invalid-name.
include-naming-hint=no

# Naming style matching correct inline iteration names.
inlinevar-naming-style=snake_case

# Regular expression matching correct inline iteration names. Overrides
# inlinevar-naming-style.
#inlinevar-rgx=

# Naming style matching correct method names.
method-naming-style=camelCase

# Regular expression matching correct method names. Overrides method-naming-
# style.
#method-rgx=

# Naming style matching correct module names.
module-naming-style=snake_case

# Regular expression matching correct module names. Overrides module-naming-
# style.
#module-rgx=

# Colon-delimited sets of names that determine each other's naming style when
# the name regexes allow several styles.
name-group=

# Regular expression which should only match function or class names that do
# not require a docstring.
no-docstring-rgx=^_

# List of decorators that produce properties, such as abc.abstractproperty. Add
# to this list to register other decorators that produce valid properties.
# These decorators are taken in consideration only for invalid-name.
property-classes=abc.abstractproperty

# Naming style matching correct variable names.
variable-naming-style=snake_case

# Regular expression matching correct variable names. Overrides variable-
# naming-style.
#variable-rgx=


[STRING]

# This flag controls whether inconsistent-quotes generates a warning when the
# character used as a quote delimiter is used inconsistently within a module.
check-quote-consistency=no

# This flag controls whether the implicit-str-concat should generate a warning
# on implicit string concatenation in sequences defined over several lines.
check-str-concat-over-line-jumps=no


[SIMILARITIES]

# Comments are removed from the similarity computation
ignore-comments=yes

# Docstrings are removed from the similarity computation
ignore-docstrings=yes

# Imports are removed from the similarity computation
ignore-imports=no

# Signatures are removed from the similarity computation
ignore-signatures=no

# Minimum lines number of a similarity.
min-similarity-lines=4


[MISCELLANEOUS]

# List of note tags to take in consideration, separated by a comma.
notes=FIXME,
      XXX,
      TODO

# Regular expression of note tags to take in consideration.
#notes-rgx=


[LOGGING]

# The type of string formatting that logging methods do. `old` means using %
# formatting, `new` is for `{}` formatting.
logging-format-style=old

# Logging modules to check that the string format arguments are in logging
# function parameter format.
logging-modules=logging


[VARIABLES]

# List of additional names supposed to be defined in builtins. Remember that
# you should avoid defining new builtins when possible.
additional-builtins=

# Tells whether unused global variables should be treated as a violation.
allow-global-unused-variables=yes

# List of names allowed to shadow builtins
allowed-redefined-builtins=

# List of strings which can identify a callback function by name. A callback
# name must start or end with one of those strings.
callbacks=cb_,
          _cb

# A regular expression matching the name of dummy variables (i.e. expected to
# not be used).
dummy-variables-rgx=_+$|(_[a-zA-Z0-9_]*[a-zA-Z0-9]+?$)|dummy|^ignored_|^unused_

# Argument names that match this expression will be ignored. Default to name
# with leading underscore.
ignored-argument-names=_.*|^ignored_|^unused_

# Tells whether we should check for unused import in __init__ files.
init-import=no

# List of qualified module names which can have objects that can redefine
# builtins.
redefining-builtins-modules=six.moves,past.builtins,future.builtins,builtins,io


[CLASSES]

# Warn about protected attribute access inside special methods
check-protected-access-in-special-methods=no

# List of method names used to declare (i.e. assign) instance attributes.
defining-attr-methods=__init__,
                      __new__,
                      setUp,
                      __post_init__

# List of member names, which should be excluded from the protected access
# warning.
exclude-protected=_asdict,
                  _fields,
                  _replace,
                  _source,
                  _make

# List of valid names for the first argument in a class method.
valid-classmethod-first-arg=cls

# List of valid names for the first argument in a metaclass class method.
valid-metaclass-classmethod-first-arg=cls


[IMPORTS]

# List of modules that can be imported at any level, not just the top level
# one.
allow-any-import-level=

# Allow wildcard imports from modules that define __all__.
allow-wildcard-with-all=no

# Analyse import fallback blocks. This can be used to support both Python 2 and
# 3 compatible code, which means that the block might have code that exists
# only in one or another interpreter, leading to false positives when analysed.
analyse-fallback-blocks=no

# Deprecated modules which should not be used, separated by a comma.
deprecated-modules=

# Output a graph (.gv or any supported image format) of external dependencies
# to the given file (report RP0402 must not be disabled).
ext-import-graph=

# Output a graph (.gv or any supported image format) of all (i.e. internal and
# external) dependencies to the given file (report RP0402 must not be
# disabled).
import-graph=

# Output a graph (.gv or any supported image format) of internal dependencies
# to the given file (report RP0402 must not be disabled).
int-import-graph=

# Force import order to recognize a module as part of the standard
# compatibility libraries.
known-standard-library=

# Force import order to recognize a module as part of a third party library.
known-third-party=enchant

# Couples of modules and preferred modules, separated by a comma.
preferred-modules=


[DESIGN]

# List of regular expressions of class ancestor names to ignore when counting
# public methods (see R0903)
exclude-too-few-public-methods=

# List of qualified class names to ignore when counting class parents (see
# R0901)
ignored-parents=

# Maximum number of arguments for function / method.
max-args=5

# Maximum number of attributes for a class (see R0902).
max-attributes=7

# Maximum number of boolean expressions in an if statement (see R0916).
max-bool-expr=5

# Maximum number of branch for function / method body.
max-branches=12

# Maximum number of locals for function / method body.
max-locals=15

# Maximum number of parents for a class (see R0901).
max-parents=7

# Maximum number of public methods for a class (see R0904).
max-public-methods=20

# Maximum number of return / yield for function / method body.
max-returns=6

# Maximum number of statements in function / method body.
max-statements=50

# Minimum number of public methods for a class (see R0903).
min-public-methods=2


[EXCEPTIONS]

# Exceptions that will emit a warning when being caught. Defaults to
# "BaseException, Exception".
overgeneral-exceptions=BaseException,
                       Exception
```

### Command used

```shell
pylint scripts/python/pylint_test_code.py
```

### Pylint output

```shell
$ pylint --version
pylint 2.12.2
astroid 2.9.3
Python 3.8.10 (default, Mar 15 2022, 12:22:08) 
[GCC 9.4.0]

$ pylint scripts/python/pylint_test_code.py 
************* Module pylint_test_code
scripts/python/pylint_test_code.py:49:0: C0301: Line too long (124/100) (line-too-long)
scripts/python/pylint_test_code.py:54:0: C0301: Line too long (143/100) (line-too-long)
scripts/python/pylint_test_code.py:79:0: C0301: Line too long (145/100) (line-too-long)
scripts/python/pylint_test_code.py:87:0: C0301: Line too long (127/100) (line-too-long)
scripts/python/pylint_test_code.py:135:0: C0301: Line too long (133/100) (line-too-long)
scripts/python/pylint_test_code.py:141:0: C0301: Line too long (132/100) (line-too-long)
scripts/python/pylint_test_code.py:8:0: C0103: Function name "module_function" doesn't conform to camelCase naming style (invalid-name)
scripts/python/pylint_test_code.py:16:0: C0103: Function name "ModuleFunction" doesn't conform to camelCase naming style (invalid-name)
scripts/python/pylint_test_code.py:26:4: C0103: Class constant name "snake_case" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/python/pylint_test_code.py:27:4: C0103: Class constant name "_snake_case" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/python/pylint_test_code.py:28:4: C0103: Class constant name "__snake_case" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/python/pylint_test_code.py:30:4: C0103: Class constant name "camelCase" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/python/pylint_test_code.py:31:4: C0103: Class constant name "_camelCase" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/python/pylint_test_code.py:32:4: C0103: Class constant name "__camelCase" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/python/pylint_test_code.py:34:4: C0103: Class constant name "PascalCase" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/python/pylint_test_code.py:35:4: C0103: Class constant name "_PascalCase" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/python/pylint_test_code.py:36:4: C0103: Class constant name "__PascalCase" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/python/pylint_test_code.py:28:4: W0238: Unused private member `SampleEnum.__snake_case` (unused-private-member)
scripts/python/pylint_test_code.py:32:4: W0238: Unused private member `SampleEnum.__camelCase` (unused-private-member)
scripts/python/pylint_test_code.py:36:4: W0238: Unused private member `SampleEnum.__PascalCase` (unused-private-member)
scripts/python/pylint_test_code.py:100:8: C0103: Attribute name "camelCaseAttr" doesn't conform to snake_case naming style (invalid-name)
scripts/python/pylint_test_code.py:101:8: C0103: Attribute name "_camelCaseAttr" doesn't conform to snake_case naming style (invalid-name)
scripts/python/pylint_test_code.py:102:8: C0103: Attribute name "__camelCaseAttr" doesn't conform to snake_case naming style (invalid-name)
scripts/python/pylint_test_code.py:104:8: C0103: Attribute name "PascalCaseAttr" doesn't conform to snake_case naming style (invalid-name)
scripts/python/pylint_test_code.py:105:8: C0103: Attribute name "_PascalCaseAttr" doesn't conform to snake_case naming style (invalid-name)
scripts/python/pylint_test_code.py:106:8: C0103: Attribute name "__PascalCaseAttr" doesn't conform to snake_case naming style (invalid-name)
scripts/python/pylint_test_code.py:39:0: R0902: Too many instance attributes (9/7) (too-many-instance-attributes)
scripts/python/pylint_test_code.py:44:4: C0103: Class attribute name "class_attr_snake_case" doesn't conform to PascalCase naming style (invalid-name)
scripts/python/pylint_test_code.py:45:4: C0103: Class attribute name "_class_attr_snake_case" doesn't conform to PascalCase naming style (invalid-name)
scripts/python/pylint_test_code.py:46:4: C0103: Class attribute name "__class_attr_snake_case" doesn't conform to PascalCase naming style (invalid-name)
scripts/python/pylint_test_code.py:48:4: C0103: Class attribute name "classAttrCamelCase" doesn't conform to PascalCase naming style (invalid-name)
scripts/python/pylint_test_code.py:50:4: C0103: Class attribute name "__classAttrCamelCase" doesn't conform to PascalCase naming style (invalid-name)
scripts/python/pylint_test_code.py:54:4: C0103: Class attribute name "__ClassAttrPascalCase" doesn't conform to PascalCase naming style (invalid-name)
scripts/python/pylint_test_code.py:59:4: C0103: Method name "class_method_snake_case" doesn't conform to camelCase naming style (invalid-name)
scripts/python/pylint_test_code.py:63:4: C0103: Method name "_class_method_snake_case" doesn't conform to camelCase naming style (invalid-name)
scripts/python/pylint_test_code.py:67:4: C0103: Method name "__class_method_snake_case" doesn't conform to camelCase naming style (invalid-name)
scripts/python/pylint_test_code.py:79:4: C0103: Method name "__classMethodCamelCase" doesn't conform to camelCase naming style (invalid-name)
scripts/python/pylint_test_code.py:83:4: C0103: Method name "ClassMethodPascalCase" doesn't conform to camelCase naming style (invalid-name)
scripts/python/pylint_test_code.py:91:4: C0103: Method name "__ClassMethodPascalCase" doesn't conform to camelCase naming style (invalid-name)
scripts/python/pylint_test_code.py:108:4: C0103: Method name "snake_case_method" doesn't conform to camelCase naming style (invalid-name)
scripts/python/pylint_test_code.py:123:4: C0103: Method name "_snake_case_method" doesn't conform to camelCase naming style (invalid-name)
scripts/python/pylint_test_code.py:126:4: C0103: Method name "__snake_case_method" doesn't conform to camelCase naming style (invalid-name)
scripts/python/pylint_test_code.py:135:4: C0103: Method name "__camelCaseMethod" doesn't conform to camelCase naming style (invalid-name)
scripts/python/pylint_test_code.py:138:4: C0103: Method name "PascalCaseMethod" doesn't conform to camelCase naming style (invalid-name)
scripts/python/pylint_test_code.py:144:4: C0103: Method name "__PascalCaseMethod" doesn't conform to camelCase naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 4.58/10 (previous run: 4.58/10, +0.00)
```

### Expected behavior

see above code marked with `ERROR:`

### Pylint version

```shell
$ pylint --version
pylint 2.12.2
astroid 2.9.3
Python 3.8.10 (default, Mar 15 2022, 12:22:08) 
[GCC 9.4.0]
```

### OS / Environment

Ubuntu 20.04

### Additional dependencies

none

**Comment by jacobtylerwalls** [comment]
Thanks for the report. As I understand it, you're reporting these inconsistencies:
- False negatives for `_single_leading_underscore` when the first letter has the wrong case given a setting for camelCase, PascalCase, etc.
- False positives in all cases for `__double_leading_underscore`, which is OK as a PEP8 "[name mangling](https://peps.python.org/pep-0008/#designing-for-inheritance)" name

These could probably be solved separately.
